daemons/attrd/attrd_commands.c

```diff
@@ -955,29 +885,141 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
+        crm_warn("Could not update %s: peer did not specify value for delay",
+                 attr);
+        return pcmk_rc_error;
+    }
+
+    dampen = crm_get_msec(dvalue);
+    if (dampen < 0) {
+        crm_warn("Could not update %s: invalid delay value %dms (%s)",
+                 attr, dampen, dvalue);
+        return pcmk_rc_error;
+    }
+
+    if (a->timeout_ms != dampen) {
+        mainloop_timer_del(a->timer);
+        a->timeout_ms = dampen;
+        if (dampen > 0) {
+            a->timer = mainloop_timer_add(attr, a->timeout_ms, FALSE,
+                                          attribute_timer_cb, a);
+            crm_info("Update attribute %s delay to %dms (%s)",
+                     attr, dampen, dvalue);
+        } else {
+            a->timer = NULL;
+            crm_info("Update attribute %s to remove delay", attr);
+        }
+
+        /* If dampening changed, do an immediate write-out,
+         * otherwise repeated dampening changes would prevent write-outs
+         */
+        write_or_elect_attribute(a);
+    }
+
+    return pcmk_rc_ok;
+}
+
+static void
+update_minimum_protocol_ver(const char *value)
+{
+    int ver;
+
+    pcmk__scan_min_int(value, &ver, 0);
+
+    if (ver > 0 && (minimum_protocol_version == -1 || ver < minimum_protocol_version)) {
+        minimum_protocol_version = ver;
+        crm_trace("Set minimum attrd protocol version to %d",
+                  minimum_protocol_version);
+    }
+}
+
+static attribute_t *
+populate_attribute(xmlNode *xml, const char *attr)
+{
+    attribute_t *a = NULL;
+    bool update_both = false;
+
+    const char *op = crm_element_value(xml, PCMK__XA_TASK);
+
+    if (attr == NULL) {
+        crm_warn("Could not update attribute: peer did not specify name");
+        return NULL;
+    }
+
+    // NULL because PCMK__ATTRD_CMD_SYNC_RESPONSE has no PCMK__XA_TASK
+    update_both = pcmk__str_eq(op, PCMK__ATTRD_CMD_UPDATE_BOTH,
+                               pcmk__str_null_matches | pcmk__str_casei);
+
+    // Look up or create attribute entry
+    a = g_hash_table_lookup(attributes, attr);
+    if (a == NULL) {
+        if (update_both || pcmk__str_eq(op, PCMK__ATTRD_CMD_UPDATE, pcmk__str_casei)) {
+            a = create_attribute(xml);
+        } else {
+            crm_warn("Could not update %s: attribute not found", attr);
+            return NULL;
+        }
+    }
+
+    // Update attribute dampening
+    if (update_both || pcmk__str_eq(op, PCMK__ATTRD_CMD_UPDATE_DELAY, pcmk__str_casei)) {
+        int rc = update_attr_dampening(a, xml, attr);
+
+        if (rc != pcmk_rc_ok || !update_both) {
+            return NULL;
+        }
+    }
+
+    return a;
+}
+
+void
+attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
+{
+    attribute_t *a = NULL;
+    const char *attr = crm_element_value(xml, PCMK__XA_ATTR_NAME);
+    const char *value = crm_element_value(xml, PCMK__XA_ATTR_VALUE);
+    gboolean is_force_write = FALSE;
```

Comment:
this is a case where int/0/1 is appropriate for a boolean since we're passing it to a function that uses int*

Reply:
Fixed.